pkg/pubsub/pubsub_client.go

```diff
@@ -0,0 +1,123 @@
+package pubsub
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/dapr"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/rabbitmq"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	flag.Var(pubSubs, "pub-sub-tool", "Preferred pub-sub tool, e.g dapr, rabbitmq. This flag can be declared more than once. Omitting will default to not using any pub sub tool.")
+}
+
+var pubSubs = newPubSubSet(map[string]PubSub{
+	dapr.Name:     &dapr.Dapr{},
+	rabbitmq.Name: &rabbitmq.RabbitMQ{},
```

Comment:
We shouldn't limit to one client per tool to make sure we have flexibilities for user to route requests to different endpoints using same tool.